roles/inspec/tasks/main.yml

```diff
@@ -1,104 +1,136 @@
-    mode: 0644
-  when: inspec.controls.rhel7.enabled|default('False')|bool
-
-- name: remove inspec lock file so we can upgrade
-  file:
-    dest: /etc/inspec/host-controls/inspec.lock
-    state: absent
-
-- name: inspec attributes
-  template:
-    src: etc/inspec/host-controls/attributes.yml
-    dest: /etc/inspec/host-controls/attributes.yml
-    mode: 0644
-
-- name: inspec sensu hook
-  sensu_check: name=inspec-check plugin=check-inspec.rb
-               args='--controls /etc/inspec/host-controls --attrs /etc/inspec/host-controls/attributes.yml' use_sudo=true
-               interval={{ inspec.interval }}
-  notify: restart sensu-client
-  when: inspec.enabled|default(True)|bool
+  - name: fetch inspec dependencies when install method is tar
+    get_url:
+      url: '{{ inspec.profiles[item].url }}'
+      dest: '/etc/inspec/profiles/archive/{{ inspec.profiles[item].name}}-{{ inspec.profiles[item].version }}.tgz'
+      mode: 0644
+    with_items: "{{ install_inspec_profiles }}"
+    when: inspec.dependency_install_method == 'tar'
+
+  - name: untar inspec dependencies
+    unarchive:
+      src: '/etc/inspec/profiles/archive/{{ inspec.profiles[item].name }}-{{ inspec.profiles[item].version }}.tgz'
+      dest: '/etc/inspec/profiles/'
+      copy: no
+    with_items: "{{ install_inspec_profiles }}"
+    when: inspec.dependency_install_method == 'tar'
+
+  - name: create inspec dependency current symlink
+    file:
+      src: "/etc/inspec/profiles/{{ inspec.profiles[item].name }}-{{ inspec.profiles[item].version }}"
+      dest: "/etc/inspec/profiles/{{ inspec.profiles[item].name }}-current"
+      state: link
+      force: true
+    with_items: "{{ install_inspec_profiles }}"
+    when: inspec.dependency_install_method == 'tar'
+
+  - name: create path for inspec control
+    file:
+      dest: "/etc/inspec/{{ item }}"
+      mode: 0755
+      state: directory
+    with_items: "{{ install_inspec_controls }}"
+
+  - name: create inspec control dir
+    file:
+      dest: "/etc/inspec/{{ item }}/controls"
+      mode: 0755
+      state: directory
+    with_items: "{{ install_inspec_controls }}"
+
+  - name: create inspec control dir
+    file:
+      dest: "/etc/inspec/{{ item }}/libraries"
+      mode: 0755
+      state: directory
+    with_items: "{{ install_inspec_controls }}"
+
+
+  - name: inspec yml for control
+    template:
+      src: etc/inspec/host-controls/inspec.yml
+      dest: /etc/inspec/{{ item }}/inspec.yml
+      mode: 0644
+    with_items: "{{ install_inspec_controls }}"
+
+  - name: control files
+    template:
+      src: etc/inspec/host-controls/controls/control.rb
+      dest: /etc/inspec/{{ item }}/controls/control.rb
+      mode: 0644
+    with_items: "{{ install_inspec_controls }}"
+
+  - name: inspec attributes
+    template:
+      src: etc/inspec/host-controls/attributes.yml
+      dest: /etc/inspec/attributes.yml
+      mode: 0644
+
+  #- name: inspec openstack control files
```

Comment:
nit: Is this dead code going to be removed or are we leaving it here for other reasons?